### DISCUSSION TITLE
Missing related consumer

### DISCUSSION THREAD

**Discussion by kingzus** [apache/apisix#8355] [General]
https://github.com/apache/apisix-docker/tree/master/example 
https://apisix.apache.org/docs/docker/example/#configure
Following the above example to launch an apisix via docker-compose, but request with an error message like "Missing related consumer".
Once I create a consumer with basic-auth at least, there  is another error {"message":"Missing authorization in request"}

How to deal with it?
route:
`{
  "uri": "/*",
  "name": "route-2",
  "methods": [
    "GET",
    "POST",
    "PUT",
    "DELETE",
    "PATCH",
    "HEAD",
    "OPTIONS",
    "CONNECT",
    "TRACE"
  ],
  "host": "web1.nijia.com",
  "service_id": "434888896975733453",
  "upstream_id": "434770829968933581",
  "labels": {
    "API_VERSION": "v2"
  },
  "status": 1
}`
services:
`{
  "name": "nijia",
  "desc": "test",
  "upstream_id": "434770829968933581"
}`

` curl  -i -X GET "http://127.0.0.1:9080" -H "Host: web1.nijia.com"               
HTTP/1.1 401 Unauthorized
Date: Fri, 18 Nov 2022 03:56:48 GMT
Content-Type: text/plain; charset=utf-8
Transfer-Encoding: chunked
Connection: keep-alive
WWW-Authenticate: Basic realm='.'
Server: APISIX/3.0.0

{"message":"Missing authorization in request"}`

**Comment by soulbird** [comment]
I didn't see your consumer information, can you complete the configuration information.

**Reply by kingzus** [reply]
{
  "username": "test",
  "desc": "test",
  "plugins": {
    "basic-auth": {
      "disable": false,
      "password": "test",
      "username": "kenvin"
    }
  }
}
Actcually, I don't want a  consumer.  I know using the following command could get the right response.
` curl  -ukenvin:test -i -X GET "http://127.0.0.1:9080" -H "Host: web1.nijia.com"`
My question the consumer is must be created?

**Reply by soulbird** [reply]
If you don't need the consumer then remove it

**Reply by kingzus** [reply]
My real purpose is direct access to web URLs without credentials. Would you mind sharing with me how to do it?

**Reply by soulbird** [reply]
Yes, can you describe each step of your configuration operation first?

**Comment by kingzus** [comment]
without a consumer, there is an error message like "Missing related consumer".

**Comment by kingzus** [comment]
It's not clear what's going on, but after clearing all the configuration and re-creating the configuration through the apisix API, the problem does not recap.

**Comment by ctxinf** [comment]
same error, config conditions:
1. multi route with same domian/path, distinguish  by `Advanced Routing Matching Conditions`
2. one route with `basic-auth` plugin

 this problem will not appear when disable the  `basic-auth`  plugin
